iree/compiler/Dialect/HAL/Transforms/MaterializeInterfaces.cpp

```diff
@@ -596,15 +597,43 @@ FuncOp InterfaceBuilder::buildRegionFuncOp() {
           offset = entryBuilder.createOrFold<mlir::ConstantIndexOp>(
               clonedFuncOp.getLoc(), value.bindingOffset.staticOffset);
         }
+
         auto bindingSymRefAttr = entryBuilder.getSymbolRefAttr(
             interfaceOp.sym_name(),
             {entryBuilder.getSymbolRefAttr(value.bindingOp)});
+
+        SmallVector<Value, 4> dynamicDims;
+        auto blockArgType = blockArg.getType().cast<Flow::DispatchTensorType>();
+        OpBuilder::InsertionGuard guard(entryBuilder);
+
+        if (!blockArgType.hasStaticShape()) {
+          // We can expect its first user to be a tie_shape op to associate
```

Comment:
It's been on my list to fix that for awhile - and this PR is a good start! If we update the flow op to carry the shape (either via a flow.func or arg attributes on the builtin.func referencing/etc) then we can (I think) remove all these tie_shapes!

Reply:
+1. This basically removes `tie_shape` ops from various backends given that `MaterializeInterfaces` is the gatekeeper for all of them. :)